packages/babel/src/index.js

```diff
@@ -0,0 +1,221 @@
+        return babel.buildExternalHelpers(null, 'module');
+      },
+
+      transform(code, filename) {
+        if (!filter(filename)) return null;
+        if (filename === HELPERS) return null;
+
+        return transformCode(
+          code,
+          { ...babelOptions, filename },
+          overrides,
+          customOptions,
+          this,
+          async (transformOptions) => {
+            if (!skipPreflightCheck) {
+              await preflightCheck(this, babelHelpers, transformOptions);
+            }
+
+            return babelHelpers === BUNDLED
+              ? addBabelPlugin(transformOptions, bundledHelpersPlugin)
+              : transformOptions;
+          }
+        );
+      }
+    };
+  };
+}
+
+function getRecommendedFormat(rollupFormat) {
+  switch (rollupFormat) {
+    case 'amd':
+      return 'amd';
+    case 'iife':
+    case 'umd':
+      return 'umd';
+    case 'system':
+      return 'systemjs';
+    default:
+      return '<module format>';
+  }
+}
+
+function createBabelOutputPluginFactory(customCallback = returnObject) {
+  const overrides = customCallback(babel);
+
+  return (pluginOptions) => {
+    const { customOptions, pluginOptionsWithOverrides } = getOptionsWithOverrides(
+      pluginOptions,
+      overrides
+    );
+
+    return {
+      name: 'babel',
+
+      renderStart(outputOptions) {
+        const { extensions, include, exclude, allowAllFormats } = pluginOptionsWithOverrides;
+
+        if (extensions || include || exclude) {
+          warnOnce(
+            this,
+            'The "include", "exclude" and "extensions" options are ignored when transforming the output.'
+          );
+        }
+        if (!allowAllFormats && outputOptions.format !== 'es' && outputOptions.format !== 'cjs') {
+          this.error(
+            `Using Babel on the generated chunks is strongly discouraged for formats other than "esm" or "cjs" as it can easily break wrapper code and lead to accidentally created global variables. Instead, you should set "output.format" to "esm" and use Babel to transform to another format, e.g. by adding "presets: [['@babel/env', { modules: '${getRecommendedFormat(
+              outputOptions.format
+            )}' }]]" to your Babel options. If you still want to proceed, add "allowAllFormats: true" to your plugin options.`
+          );
+        }
+      },
+
+      renderChunk(code, chunk, outputOptions) {
+        /* eslint-disable no-unused-vars */
+        const {
+          allowAllFormats,
+          exclude,
+          extensions,
+          externalHelpers,
+          externalHelpersWhitelist,
+          include,
+          runtimeHelpers,
+          ...babelOptions
+        } = unpackOutputPluginOptions(pluginOptionsWithOverrides, outputOptions);
+        /* eslint-enable no-unused-vars */
+
+        return transformCode(code, babelOptions, overrides, customOptions, this);
+      }
+    };
+  };
+}
+
+const babelPluginFactory = createBabelInputPluginFactory();
```

Comment:
I would gladly restructure this - I believe those should be 4 separate exports, rather than a single export with static properties. This PR will land v5 of this plugin so this is IMHO OK to do.

WDYT about those:
- default export (result of `createBabelInputPluginFactory()`)
- `createBabelInputPluginFactory`
- `babelOutputPluginFactory`
- `createBabelOutputPluginFactory`

Those named exports would be quite mouthful 🤷‍♂ but correct. I'm open for other names - but cant figure out anything better myself.

Reply:
IMO the `createXY` names are ok as they will not be used often and only by people who know what they are doing. However I think functions should start with a verb, and the return value of `createBabelOutputPluginFactory` should be called `getOutputPlugin`, which also sounds less intimidating. For symmetry, I would furthermore export the default export additionally as  `getInputPlugin`. This will make @TrySound happy 😃

Reply:
On second thought `getBabelOutputPlugin` and `getBabelInputPlugin` so that it does not conflict with other plugins

Reply:
Also note that using named and default export together will switch the plugin to named-exports-mode when imported from CJS. This means that `const babel = require(‘rollup-plugin-babel’)` will no longer work. And I am not sure if the new CLI `rollup -c —plugin babel` would work. The latter could be fixed by providing a named export `babel` instead of a default. In any case, this change would be breaking.

Reply:
I've changed exports to pretty much to what you have suggested, you can check out this commit: https://github.com/rollup/plugins/pull/108/commits/4c5e1a8bf885d701af5d87edc4b595458870dcf9